src/GenericHostBuilderExample/Program.cs

```diff
@@ -1,20 +1,45 @@
 ﻿using System.Threading.Tasks;
 using Autofac;
 using Autofac.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection;
 using Microsoft.Extensions.Hosting;
 
 namespace GenericHostBuilderExample
 {
     internal class Program
     {
-        private static async Task Main() =>
-            await new HostBuilder()
+        public static async Task Main(string[] args)
+        {
+            // The `UseServiceProviderFactory(new AutofacServiceProviderFactory())` call here allows for
+            // ConfigureContainer to be supported with
+            // a strongly-typed ContainerBuilder. If you don't
+            // have the call to `UseServiceProviderFactory(new AutofacServiceProviderFactory())` here, you won't get
+            // ConfigureContainer support. This also automatically
+            // calls Populate to put services you register during
+            // ConfigureServices into Autofac.
+            await Host.CreateDefaultBuilder(args)
                 .UseServiceProviderFactory(new AutofacServiceProviderFactory())
-                .ConfigureContainer<ContainerBuilder>((context, builder) =>
-                {
-                    builder.RegisterType<Logger>().As<ILogger>();
-                    builder.RegisterType<HostedService>().As<IHostedService>();
-                })
+                .ConfigureServices(ConfigureServices)
+                .ConfigureContainer<ContainerBuilder>(ConfigureContainer)
                 .RunConsoleAsync();
+        }
+
+
+        private static void ConfigureServices(IServiceCollection services)
+            // Use extensions from libraries to register services in the
+            // collection. These will be automatically added to the
+            // Autofac container.
+            => services.AddHostedService<HostedService>();
```

Comment:
I haven't messed much with the new generic host. Is `AddHostedService` from that? I don't see it in the new code, maybe I'm missing it.

Reply:
It's an extension method on `IServiceCollection` that has been there for a while. Here the [2.2.6 tag](https://github.com/aspnet/Extensions/blob/v2.2.6/src/Hosting/Abstractions/src/ServiceCollectionHostedServiceExtensions.cs). It is part of `Microsoft.Extensions.Hosting.Abstractions`

Reply:
Cool! I've never used that.